packages/components/src/mobile/bottom-sheet/bottom-sheet-navigation/navigation-screen.native.js

```diff
@@ -79,11 +80,19 @@ const BottomSheetNavigationScreen = ( {
 	};
 	return useMemo( () => {
 		return isScrollable || isNested ? (
-			<View onLayout={ onLayout }>{ children }</View>
+			<View
+				onLayout={ onLayout }
+				testID={ `navigation-screen-${ name }` }
```

Comment:
Hmm, feels to me that this comment is useful to have as an inline comment in the code itself? I mean, it's not trivial to tell that we're using it here to sidestep the layout optimization algorithms, right?

Reply:
> I mean, it's not trivial to tell that we're using it here to sidestep the layout optimization algorithms, right?

@hypest I would not say we are "sidestepping layout optimization algorithms." There are two things occurring here. 

First, the element targeted has no user-facing identifier. I.e. we cannot use a less brittle, user-facing query like `getByText` to find the subject element. In general, [we avoid using `testID`s unless it is necessary](https://testing-library.com/docs/queries/bytestid/): 

>In the spirit of [the guiding principles](https://testing-library.com/docs/guiding-principles), it is recommended to use this only after the other queries don't work for your use case. Using data-testid attributes do not resemble how your software is used and should be avoided if possible. That said, they are _way_ better than querying based on DOM structure or styling css class names. Learn more about `data-testids` from the blog post "[Making your UI tests resilient to change](https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change)"

Second, the reason this element is queried is because the subject code relies upon React Native's `onLayout` hook, which is [not triggered by React Native Testing Library](https://github.com/callstack/react-native-testing-library/issues/240#issuecomment-559877887), due to its roots in native layout phases. The `testID` here is utilized to manually trigger the `onLayout` hook within the test. 

My comment on the pull request was largely aimed at the first point, explaining why I queried the subject element via a less-than-ideal query. We have [used this approach before](https://github.com/WordPress/gutenberg/pull/32038/files#diff-d5f0be6da4449d0e7e208bc29d6ee352d7923f69d26c9c0965a478de0238bb3eR47-R59). However, if you feel it is worth communicating either of this points in the code, I am happy to leave an inline comment. 

WDYT?

Reply:
Oh, gotcha. I interpreted the `testIDs added to trigger layout events on elements that have no user-facing identifiers.` as an attempt to [disable the 'layout-only' view removal optimization](https://reactnative.dev/docs/view#testid). I know understand that it's only added for exposing the view to the tests.

Since this was mostly a misinterpretation of the PR comment itself, someone reading just the code won't be really confused so, I don't think there's a need for an inline comment 👍, thanks!